components/video/video.cpp

```diff
@@ -0,0 +1,252 @@
+#include "video.h"
+
+#include <faio/faio.h>
+#include <SDL.h>
+
+extern "C" {
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libavformat/avio.h>
+#include <libswscale/swscale.h>
+}
+
+namespace Video
+{
+
+AVCodec * Video::sCodec;
+AVFrame * Video::sFrame;
+AVFrame * Video::sFrameRGB;
+const int Video::BUFFER_SIZE = 8192;
+
+
+bool Video::init()
+{
+    av_register_all();
+    avcodec_register_all();
+
+    Video::sCodec       = avcodec_find_decoder(AV_CODEC_ID_SMACKVIDEO);
+    Video::sFrame       = av_frame_alloc();
+    Video::sFrameRGB    = av_frame_alloc();
+
+    return true;
+}
+
+
+void SaveFrame(AVFrame *pFrame, int width, int height, int iFrame)
+{
+    FILE *pFile;
+    char szFilename[32];
+    int  y;
+
+    // Open file
+    sprintf(szFilename, "frame%d.ppm", iFrame);
+    pFile=fopen(szFilename, "wb");
+    if(pFile==NULL)
+        return;
+
+    // Write header
+    fprintf(pFile, "P6\n%d %d\n255\n", width, height);
+
+    // Write pixel data
+    for(y=0; y<height; y++)
+        fwrite(pFrame->data[0]+y*pFrame->linesize[0], 1, width*3, pFile);
+
+    // Close file
+    fclose(pFile);
+}
+
+
+static int readFunction(void* opaque, uint8_t* buf, int buf_size) {
+
+    FAIO::FAFile& file = *reinterpret_cast<FAIO::FAFile*>(opaque);
+    size_t bytes = FAIO::FAfread(reinterpret_cast<void*>(buf), buf_size, 1, &file);
+    return bytes;
+}
+
+Video::Video(unsigned int width, unsigned int height)
+    :
+      mWidth(width),
+      mHeight(height),
+      mCurrentFrame(-1),
+      mNumFrames(0)
+{
+}
+
+Video::~Video()
+{
+    int size = mSprites.size();
+    for(int i = 0 ; i < size; i++)
+    {
+        SDL_DestroyTexture(reinterpret_cast<SDL_Texture*>(mSprites[i]));
```

Comment:
This should be handled in the render component, nothing but that should know sdl even exists